src/conn/conn_log.c

```diff
@@ -139,30 +134,21 @@ __logmgr_version(WT_SESSION_IMPL *session, bool reconfig)
         return (0);
 
     /*
-     * Set the log file format versions based on compatibility versions set in the connection. We
-     * must set this before we call log_open to open or create a log file.
+     * Set the log file format versions based on compatibility versions set in the connection. The
+     * compatibility version must be set at this point. We must set this before we call log_open to
+     * open or create a log file.
      *
      * Note: downgrade in this context means the new version is not the latest possible version. It
      * does not mean the direction of change from the release we may be running currently.
      */
-    if (conn->compat_major == WT_LOG_V5_MAJOR) {
```

Comment:
The `new_version` logic here is a dupe of `__logmgr_set_log_version` above so I've reused it.
The only difference is the lack of the `if(__wt_version_defined())` check in the original code indicating that `conn->compat_version` is always defined here. I've added a `WT_ASSERT` to account for this.